subprojects/gradle/instrumentation-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/InstrumentationTestImpactAnalysisPlugin.kt

```diff
@@ -20,66 +21,57 @@ class InstrumentationTestImpactAnalysisPlugin : Plugin<Project> {
 
     override fun apply(project: Project) {
-        require(project.isAndroid()) { "${this::class.java.simpleName} must be applied only in Android Application module type" }
-        require(!project.isRoot()) { "${this::class.java.simpleName} should be applied to the specific project, which produces test apk" }
-        require(project.rootProject.plugins.hasPlugin("com.avito.android.impact")) {
-            "${this.javaClass.simpleName} requires impact plugin applied to the project"
-        }
+        checkPreconditions(project)
 
-        val extension = project.extensions.create(
-            "instrumentationTestImpactAnalysis",
-            InstrumentationTestImpactAnalysisExtension::class.java,
-            project
-        )
+        val extension = project.extensions
+            .create<InstrumentationTestImpactAnalysisExtension>("instrumentationTestImpactAnalysis", project)
 
+        //todo should work in any environment
         if (project.buildEnvironment !is BuildEnvironment.CI) return
 
         modifiedProjectsFinder = ModifiedProjectsFinder.from(project)
         modulesBytecodeResolver = BytecodeResolver(project)
 
-        val bytecodeAnalyzeTask = project.tasks.register(
-            "analyzeTestBytecode",
-            TestBytecodeAnalyzeTask::class.java,
-            extension,
-            modifiedProjectsFinder,
-            modulesBytecodeResolver
-        )
+        project.withAndroidApp {
 
-        bytecodeAnalyzeTask.configure { task ->
-            task.timeout.set(Duration.ofMinutes(20))
-            task.group = TASK_GROUP
-            task.description = "Analyze androidTest bytecode to collect maps: [Screen:Test], [Screen:RootId]"
+            val bytecodeAnalyzeTask = project.tasks.register<TestBytecodeAnalyzeTask>(
+                "analyzeTestBytecode",
+                extension,
+                modifiedProjectsFinder,
+                modulesBytecodeResolver
+            ) {
+                timeout.set(Duration.ofMinutes(20))
+                group = TASK_GROUP
+                description = "Analyze androidTest bytecode to collect maps: [Screen:Test], [Screen:RootId]"
 
-            task.dependsOn("${project.path}:compileDebugAndroidTestKotlin") // TODO
-        }
-        project.tasks.register(
-            TASK_ANALYZE_TEST_IMPACT_ANALYSIS,
-            AnalyzeTestImpact::class.java,
-            extension,
-            modifiedProjectsFinder,
-            gson
-        ).configure { analyzeTask ->
-            analyzeTask.group = TASK_GROUP
-            analyzeTask.description = "Find tests to run based on changed modules and test code analysis"
-            analyzeTask.dependsOn(bytecodeAnalyzeTask) // TODO https://docs.gradle.org/5.6.4/userguide/lazy_configuration.html#sec:working_with_task_dependencies_in_lazy_properties
-            analyzeTask.apply {
+                dependsOn("${project.path}:compile${it.testBuildType}AndroidTestKotlin")
```

Comment:
could be only one `testBuildType`, no need to produce multiple tasks

Reply:
is it a build variant or just a build type? should we support build flavors?